www/docs/main/quickstart.mdx

```diff
@@ -10,239 +10,347 @@ import CodeBlock from '@theme/CodeBlock';
+      <td>Stuff accessible to all <b className="text-green-500">procedures</b> (e.g. session state, db connection)</td>
+    </tr>
+    <tr>
+      <td><b className="text-indigo-400">Middleware</b> <a href="/docs/server/middlewares">↗</a></td>
+      <td>Functions executed before and after <b className="text-green-500">procedures</b>, can modify <b className="text-pink-500">context</b></td>
     </tr>
-
   </tbody>
 </table>
 
-### Installation Snippets
+The last two we won't be touching on in this guide, but you can read more about them in the [middlewares](/docs/server/middlewares) and [context](/docs/server/context) docs.
+
+
+## Installation
+
+tRPC is broken up into several packages, so you can install only what you need. Make sure to install the packages you want in the proper sections of your codebase. For this quickstart guide we'll keep it simple and use the vanilla client only. For framework guides, checkout [usage with React](/docs/reactjs/setup) and [usage with Next.js](/docs/nextjs/setup).
+
+:::info Requirements
+- tRPC requires TypeScript >= 4.7.0
+- We strongly recommend you using `"strict": true` in your `tsconfig.json` as we don't officially support non-strict mode.
+:::
 
-Here are some install scripts to add tRPC to your project. These scripts include every `@trpc/*` package, so feel free to remove what you don't need!
+Start off by installing the `@trpc/server` and `@trpc/client` packages:
 
 <Tabs>
   <TabItem value="npm" label="npm" default>
 
 ```sh
-npm install @trpc/server @trpc/client @trpc/react-query @trpc/next @tanstack/react-query
+npm install @trpc/server @trpc/client
 ```
 
   </TabItem>
   <TabItem value="yarn" label="yarn">
 
 ```sh
-yarn add @trpc/server @trpc/client @trpc/react-query @trpc/next @tanstack/react-query
+yarn add @trpc/server @trpc/client
 ```
 
   </TabItem>
   <TabItem value="pnpm" label="pnpm">
 
 ```sh
-pnpm add @trpc/server @trpc/client @trpc/react-query @trpc/next @tanstack/react-query
+pnpm add @trpc/server @trpc/client
 ```
 
   </TabItem>
 </Tabs>
 
 ## Defining a backend router
 
-Let's walk through the steps of building a typesafe API with tRPC. To start, this API will only contain two endpoints with these TypeScript signatures:
+Let's walk through the steps of building a typesafe API with tRPC. To start, this API will contain three endpoints with these TypeScript signatures:
 
 ```ts
-userById(id: string) => { id: string; name: string; }
-userCreate(data: { name: string }) => { id: string; name: string; }
+type User = { id: string; name: string; };
+
+userList: () => User[];
+userById: (id: string) => User;
+userCreate: (data: { name: string }) => User;
 ```
 
-### Create a router instance
+### 1. Create a router instance
 
-First, let's define an empty router in our server codebase:
+First, let's initilize the tRPC backend. It's good convention to do this in a separate file and export reusable helper functions instead of the entire tRPC object.
 
```

Comment:
:::warning
If you choose a different file structure than we recommend, then beware of [circular dependencies](https://stackoverflow.com/questions/46589957/es6-modules-and-circular-dependency). Typically `appRouter` needs `t` and `yourOwnSubRouter`, and `yourOwnSubRouter` needs `t` too. So if `t` and `appRouter` are in the same file you may end up with a strange error like `t is undefined` at run-time, because both files are importing each other and both can't be initialised already.
:::